pages/docs/reason-compiler/latest/lazy-values.mdx

```diff
@@ -0,0 +1,139 @@
+# Lazy Values
+
+A lazy value represents a deferred computation which will automatically memoize the result on the first run, and then return the memoized result on any repeated execution.
+
+This is useful for defining functions and expressions for complex procedures that always return the same value, for example:
+
+- Doing expensive DOM traversals over the same tree over and over again
```

Comment:
This isn't what I meant by my comment on Discord. The DOM is so globally mutable so you _want_ to re-traverse most of the time.

What I'm talking about is having multiple expensive traversals and I want to use the first one that succeeds. It's easier to write this with lazy syntax than a list of functions, _particularly_ if each function takes different arguments.